### PR TITLE
Updates for latest ensembles

### PR DESCRIPTION
**Pull request by sophiemiddleton**
Includes:
- updates to ensembling scripts to allow for grid use
- pion filters for RPC work

### CODE REVIEW

**Comment by sophiemiddleton** [comment]
with https://github.com/Mu2e/Offline/pull/1341

**Comment by sophiemiddleton** [comment]
Any comments on this? I would like to start making RPC samples this week to make MDS-1

File: Scripts/CreateSimEfficiency.sh [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 #
 rm $1_SimEff.txt
 mu2eGenFilterEff --out=$1_SimEff.txt --chunksize=100 sim.mu2e.MuBeamCat.$1.art sim.mu2e.EleBeamCat.$1.art sim.mu2e.NeutralsCat.$1.art \
-sim.mu2e.MuminusStopsCat.$1.art sim.mu2e.MuplusStopsCat.$1.art sim.mu2e.IPAMuminusStopsCat.$1.art \
+sim.mu2e.MuminusStopsCat.$1.art sim.mu2e.MuplusStopsCat.$1.art sim.mu2e.IPAMuminusStopsCat.$2.art \
 dts.mu2e.MuBeamFlashCat.$1.art dts.mu2e.EleBeamFlashCat.$1.art dts.mu2e.NeutralsFlashCat.$1.art \
 dts.mu2e.MuStopPileupCat.$1.art \
-dts.mu2e.EarlyMuBeamFlashCat.$1.art dts.mu2e.EarlyEleBeamFlashCat.$1.art dts.mu2e.EarlyNeutralsFlashCat.$1.art
+dts.mu2e.EarlyMuBeamFlashCat.$1.art dts.mu2e.EarlyEleBeamFlashCat.$1.art dts.mu2e.EarlyNeutralsFlashCat.$1.art \
+sim.mu2e.PiminusStopsCat.$3.art sim.mu2e.PiTargetStops.$3.art sim.mu2e.PiMinusFilter.$4.art
```
Comment (brownd1978) [inline Scripts/CreateSimEfficiency.sh:16]: As the piminus output comes from a special production sequence I think we don't want this

File: JobConfig/primary/TargetPiStopPreFilter.fcl [hunk 2]
```diff
@@ -0,0 +1,13 @@
+//
+// Base configuration for prefiltering pions based on time
+// original author: Sophie Middleton, 2024
+//
+#include "Production/JobConfig/primary/StopParticleFilter.fcl"
```
Comment (brownd1978) [inline JobConfig/primary/TargetPiStopPreFilter.fcl:5]: Is this the only place StopParticleFilter is used?

**Review by brownd1978 — COMMENTED** [review]
We should address the question about separate SimEfficiency tables for special runs (pion is one example) with database experts.

File: Scripts/CreateSimEfficiency.sh (hunk 1 above)
Comment (sophiemiddleton) [reply Scripts/CreateSimEfficiency.sh:16]: Correct, I made a new script for pions only

File: JobConfig/primary/TargetPiStopPreFilter.fcl (hunk 2 above)
Comment (sophiemiddleton) [reply JobConfig/primary/TargetPiStopPreFilter.fcl:5]: currently I think so

**Comment by sophiemiddleton** [comment]
Any more comments on this, it would be good to have it merged and tagged in a SimJob to use the python scripts that way

**Review by brownd1978 — APPROVED** [review]

File: Scripts/removezero.sh
Comment (oksuzian) [inline Scripts/removezero.sh:1]: What's the purpose of this script?
Why is it using hardcoded versions of eventCount, col1?

File: CampaignConfig/mdc2020_RPC.ini
Comment (oksuzian) [inline CampaignConfig/mdc2020_RPC.ini:1]: I thought we moved to jitfcl?

File: CampaignConfig/mdc2024_DIOtail.ini
Comment (oksuzian) [inline CampaignConfig/mdc2024_DIOtail.ini:1]: This seems redundant.
Primaries are generated with CampaignConfig/mdc2020_primary.ini.

File: CampaignConfig/mdc2024_ensembles.cfg
Comment (oksuzian) [inline CampaignConfig/mdc2024_ensembles.cfg:1]: I suggest we don't use private includes in the Production repository.
I also propose a workflow that uses [global] in the prolog ONLY.
Having said that, I'm fine with the addition of this file. I will just need to adapt it.

File: CampaignConfig/mdc2024_ensembles.ini
Comment (oksuzian) [inline CampaignConfig/mdc2024_ensembles.ini:1]: Same as before, are we still using _fcls?

File: CampaignConfig/mdc2024_primary.cfg
Comment (oksuzian) [inline CampaignConfig/mdc2024_primary.cfg:1]: Can we use mdc2020_primary.cfg instead?

File: JobConfig/ensemble/scripts/initialscripts/getLivetime.sh
Comment (oksuzian) [inline JobConfig/ensemble/scripts/initialscripts/getLivetime.sh:1]: Avoid using the reference to a private code.

**Review by oksuzian — CHANGES_REQUESTED** [review]
Please address the suggestions

File: Scripts/removezero.sh
Comment (sophiemiddleton) [reply Scripts/removezero.sh:1]: It reads in a list of stopped pions (post filtering) it removes any file with 0 events (hence the name). It is not necessary for SAM based workflows, we would need another script. I used it for the interactive tests

**Comment by sophiemiddleton** [comment]
I have done the required changes. As for the ini/cfg for the ensembles, I propose us discussing this at the Production meeting, there's a few complexities and those scripts are mostly out of date now

**Review by oksuzian — APPROVED** [review]
I need additional info on ensemble generation.
I'm fine with the current PR as it does not collide with existing tools.

**Comment by sophiemiddleton** [comment]
@brownd1978 I think we are good to merge this. I will work on the cfg/ini files which we discussed in the meeting yesterday. Thats a spearate PR

File: Scripts/gen_Mix.sh [hunk 3]
```diff
@@ -261,6 +261,8 @@ echo services.DbService.purpose: ${CAMPAIGN}_${DBPURPOSE} >> mix.fcl
 echo services.DbService.version: ${DBVERSION} >> mix.fcl
 echo services.DbService.verbose : 2 >> mix.fcl
 echo "services.GeometryService.bFieldFile : \"${FIELD}\"" >> mix.fcl
+echo "#include \"Production/JobConfig/digitize/MakeSurfaceSteps.fcl\"">> mix.fcl
```
Comment (brownd1978) [inline Scripts/gen_Mix.sh:264]: If we're adding SurfaceSteps for mixing they shoudl also be in digitize.

Comment (sophiemiddleton) [reply Scripts/gen_Mix.sh:264]: I think this is already in the gen digi script if thats what you mean?

**Pull request merged**